projects/smoke_can/src/main.c

```diff
@@ -0,0 +1,90 @@
+// this is a simple smoke test to make sure can works
+// simply run the project and it should periodically send can messages and receive them
+
+// Configurable items: bitrate, send time, data
+#include <string.h>
+
+#include "can.h"
+#include "can_msg.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#define TEST_CAN_DEVICE_ID 0x1
+#define TEST_CAN_MSG_ID 0x1
+#define TEST_CAN_BITRATE CAN_HW_BITRATE_500KBPS
+
+// this is how often the message will send
+// modify if you want to send more or less
+#define SMOKETEST_SEND_TIME_MS 1000
+// this is the data it'll send
+// change if you wish
+#define DATA \
+  { 0xfe, 0xdc, 0xba, 0x98, 0x76, 0x54, 0x32, 0x10 }
```

Comment:
Make sure to say that it's an array of exactly 8 uint8s. In fact, it might be better to replace it with a `static const uint8_t s_data[8] = {...}` just to make it explicit.

Reply:
will do

Reply:
it keeps throwing a 
initialization makes integer from pointer without a cast
so I'll just keep it as a #define